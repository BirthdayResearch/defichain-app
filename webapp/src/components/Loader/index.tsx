import React from 'react';
import MDSpinner from 'react-md-spinner';

interface LoaderProps {
  size?: number;
  color?: string;
  className?: string;
  borderSize?: number;
}

const Loader: React.FunctionComponent<LoaderProps> = ({
  size,
  color,
  className,
  borderSize,
}) => (
  <div className={`${className || ''}`}>
    <MDSpinner
      size={size || 28}
      singleColor={color || '#666666'}
      borderSize={borderSize || 4}
    />
  </div>
);

export default Loader;
