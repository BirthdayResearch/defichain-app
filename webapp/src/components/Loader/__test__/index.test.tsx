import React from 'react';
import Loader from '../index';
import { shallow } from 'enzyme';

describe('Loader component', () => {
  it('should check for snapshot', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
