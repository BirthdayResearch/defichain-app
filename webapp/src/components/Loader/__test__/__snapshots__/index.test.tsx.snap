// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should check for snapshot 1`] = `
<div
  className=""
>
  <MDSpinner
    borderSize={4}
    color1="rgb(66, 165, 245)"
    color2="rgb(239, 83, 80)"
    color3="rgb(253, 216, 53)"
    color4="rgb(76, 175, 80)"
    duration={1333}
    singleColor="#666666"
    size={28}
  />
</div>
`;
